addons/website_payment/controllers/portal.py

```diff
@@ -0,0 +1,154 @@
+    )
+    def get_acquirer_fees(self, acquirer_ids=None, amount=None, currency_id=None, country_id=None):
+        # Cast numeric parameters as int or float and void them if their str value is malformed
+        # acquirer_ids = json.loads(acquirer_ids or '[]')
+        amount = self.cast_as_float(amount)
+        currency_id, country_id = tuple(map(
+            self.cast_as_int, (currency_id, country_id)
+        ))
+
+        acquirers_sudo = request.env['payment.acquirer'].sudo().browse(acquirer_ids)
+
+        # Make sure that the currency exists and is active
+        currency = request.env['res.currency'].browse(currency_id).exists()
+        if not currency or not currency.active:
+            raise werkzeug.exceptions.NotFound  # The currency must exist and be active
+
+        country = request.env['res.country'].browse(country_id)
+
+        # Compute the fees taken by acquirers supporting the feature
+        fees_by_acquirer = {
+            acq_sudo.id: acq_sudo._compute_fees(amount, currency, country)
+            for acq_sudo in acquirers_sudo.filtered('fees_active')
+        }
+        return fees_by_acquirer
+
+    @http.route(
+        '/donation/transaction/<minimum_amount>', type='json', methods=['POST'], auth='public', website=True, sitemap=False,
+    )
+    def donation_transaction(self, amount, currency_id, partner_id, access_token, minimum_amount=0, **kwargs):
+        if float(amount) < float(minimum_amount):
+            raise ValidationError(_('Donation amount must be at least %.2f.', float(minimum_amount)))
+        public_partner_id = request.env.ref('base.public_partner')
+        is_public_partner = partner_id == public_partner_id.id or not partner_id
+        if is_public_partner:
+            details = kwargs['partner_details']
+            if 'name' not in details or not details['name']:
+                raise ValidationError(_('Name is required.'))
+            if 'email' not in details or not details['email']:
+                raise ValidationError(_('Email is required.'))
+            partner_id = public_partner_id.id
+            del kwargs['partner_details']
+
+        kwargs.pop('custom_create_values', None)  # Don't allow passing arbitrary create values
+        tx_sudo = self._create_transaction(
+            amount=amount, currency_id=currency_id, partner_id=partner_id, **kwargs
+        )
+        tx_sudo.is_donation = True
+        if is_public_partner:
+            tx_sudo.update({
+                'partner_name': details['name'],
+                'partner_email': details['email'],
+                'partner_country_id': details['country_id'],
+            })
+        self._update_landing_routes(tx_sudo)
+
+        # Send a notification to warn that a donation has been made
+        recipient_email = kwargs['donation_recipient_email']
+        comment = kwargs['donation_comment']
+        tx_sudo._send_donation_email(True, comment, recipient_email)
```

Comment:
As discussed the other day, the transaction is not confirmed nor even initiated. All you know is that the customer clicked the "Donate" button.

Reply:
Indeed, but after verification this is the wanted behavior: be informed that somebody thought about making a donation so that the person could be re-contacted manually in case the donation did not actually occur.